dali/dfu/dfurun.cpp

```diff
@@ -533,6 +533,24 @@ class CDFUengine: public CInterface, implements IDFUengine
         }
     }
 
+    void ensureAFilePermissions(IFileDescriptor * fd, SecAccessFlags perm, bool write)
+    {
+        StringBuffer traceName;
+        fd->getTraceName(traceName);
+        if (traceName.length() > 255)
+        {
+            traceName.setLength(255);
+            traceName.append("...");
+        }
```

Comment:
Better to only use the above block if needed (if denied).
I'd go with:
```
if ((write && !HASWRITEPERMISSION(perm)) || (!write && !HASREADPERMISSION(perm)))
{
    StringBuffer traceName;
    fd->getTraceName(traceName);
    if (traceName.length() > 255)
    {
        traceName.setLength(255);
        traceName.append("...");
    }
    if (write)
        throw makeStringExceptionV(DFSERR_CreateAccessDenied, "Create permission denied for physical file(s): %s", traceName.str());
    else
        throw makeStringExceptionV(DFSERR_LookupAccessDenied, "Lookup permission denied for physical file(s): %s", traceName.str());
}
```

Reply:
This function may be used by the checkForeignFilePermissions() in https://github.com/hpcc-systems/HPCC-Platform/pull/17563. I may change it to:
```
void ensureFilePermissions(const char * fileName, IFileDescriptor * fd, SecAccessFlags perm, bool write)
    {
        if ((write && !HASWRITEPERMISSION(perm)) || (!write && !HASREADPERMISSION(perm)))
        {
            StringBuffer traceName;
            if (isEmptyString(fileName))
            {
                fd->getTraceName(traceName);
                elideString(traceName, 255);
                fileName = traceName.str();
            }
            if (write)
                throw makeStringExceptionV(DFSERR_CreateAccessDenied, "Create permission denied for physical file(s): %s", fileName);
            else
                throw makeStringExceptionV(DFSERR_LookupAccessDenied, "Lookup permission denied for physical file(s): %s", fileName);
        }
    }
```

Reply:
Also add the ensureFilePermissions() into https://github.com/hpcc-systems/HPCC-Platform/pull/17563.